src/create-embed.js

```diff
@@ -0,0 +1,283 @@
+  inner.appendChild(el);
+
+  const outer = document.createElement('div');
+
+  outer.style.position = 'relative';
+  outer.style.display = 'block';
+  outer.style.maxWidth = maxWidth;
+  outer.appendChild(inner);
+
+  return outer;
+};
+
+/**
+ * Wraps an element in a Picture-in-Picture plugin container.
+ *
+ * @private
+ * @param  {Object} embedOptions
+ *         Embed options from the params.
+ *
+ * @param  {Element} el
+ *         The DOM element.
+ *
+ * @return {Element}
+ *         A new element (if needed).
+ */
+const wrapPip = (embedOptions, el) => {
+  if (!embedOptions.pip) {
+    return el;
+  }
+
+  const pip = document.createElement('div');
+
+  pip.classList.add('vjs-pip-container');
+  pip.appendChild(el);
+
+  return pip;
+};
+
+/**
+ * Wraps a bare embed element with necessary parent elements, depending on
+ * embed options given in params.
+ *
+ * @private
+ * @param  {Object} embedOptions
+ *         Embed options from the params.
+ *
+ * @param  {Element} embed
+ *         The embed DOM element.
+ *
+ * @return {Element}
+ *         A new element (if needed) or the embed itself.
+ */
+const wrapEmbed = (embedOptions, embed) => {
+  if (!embedOptions) {
+    return embed;
+  }
+
+  return wrapPip(embedOptions, wrapResponsive(embedOptions, embed));
+};
+
+/**
+ * Inserts a previously-created embed element into the page based on params.
+ *
+ * @private
+ * @param  {Object} params
+ *         A parameters object. See README for details.
+ *
+ * @param  {Element} embed
+ *         The embed DOM element.
+ *
+ * @return {Element}
+ *         The embed DOM element.
+ */
+const insertEmbed = (params, embed) => {
+  const {refNodeInsert} = params;
+  let {refNode} = params;
+
+  if (typeof refNode === 'string') {
+    refNode = document.querySelector(refNode);
+  }
+
+  if (!refNode || refNode.nodeType !== 1 || !refNode.parentNode) {
+    throw new Error('missing/invalid refNode');
+  }
+
+  // Wrap the embed, if needed, in container elements to support various
+  // plugins.
+  const wrapped = wrapEmbed(params.embedOptions, embed);
+
+  // Decide where to insert the wrapped embed.
+  if (refNodeInsert === REF_NODE_INSERT_PREPEND) {
+    if (refNode.firstChild) {
+      refNode.insertBefore(wrapped, refNode.firstChild);
+    } else {
+      refNode.appendChild(wrapped);
+    }
+  } else if (refNodeInsert === REF_NODE_INSERT_APPEND) {
+    refNode.appendChild(wrapped);
+  } else if (refNodeInsert === REF_NODE_INSERT_BEFORE) {
+    refNode.parentNode.insertBefore(wrapped, refNode);
+  } else if (refNodeInsert === REF_NODE_INSERT_AFTER) {
+    insertAfter(wrapped, refNode);
+  } else {
+    refNode.parentNode.replaceChild(wrapped, refNode);
+  }
+
+  // If the playlist embed option is provided, we need to add a playlist element
+  // immediately after the embed. This has to happen after the embed is inserted
+  // into the DOM (above).
+  if (params.embedOptions && params.embedOptions.playlist) {
```

Comment:
Yeah. When I started working on this feature, I was going to allow you to configure the element type - `ol` or `div` for different versions of the playlist plugins - but decided that was probably overkill since we want people using the new stuff, anyway.